java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessagingFactory.java

```diff
@@ -136,14 +137,17 @@ private void startReactor(ReactorHandler reactorHandler) throws IOException
 	}
 
 	@Override
-	public Connection getConnection()
+	public Session getSession(final String sessionId)
 	{
 		if (this.connection == null || this.connection.getLocalState() == EndpointState.CLOSED || this.connection.getRemoteState() == EndpointState.CLOSED)
 		{
 			this.connection = this.getReactor().connectionToHost(this.hostName, ClientConstants.AMQPS_PORT, this.connectionHandler);
 		}
 
-		return this.connection;
+		final Session session = this.connection.session();
+		session.open();
+		
+		return session;
```

Comment:
This deosn't mean that *All* links share same session.

Since, request-response links share the same session - this is an infrastructure to let links opt-in to share a session (based on SessionId). If the sessionId is null - then a new Session will be created.

---
In reply to: [78790801](https://github.com/Azure/azure-event-hubs/pull/260#discussion_r78790801) [](ancestors = 78790801)